src/Test.UnitTests.Sarif/Core/ResultTests.cs

```diff
@@ -43,6 +48,95 @@ public void Result_Level_ResetsKindValue()
             result.Kind.Should().Be(ResultKind.Fail);
         }
 
+        [Fact]
+        public void Result_Level_WriteCorrectValue_GeneralCases()
+        {
+            Result result;
+            ReportingConfiguration defaultConfiguration;
+            ReportingConfiguration overrideConfiguration;
+
+            result = new Result() { Level = FailureLevel.Error };
+            Assert.True(WriteSarifThenReadLevelNode(result) == FailureLevel.Error);
+
+            result = new Result() { Level = FailureLevel.Note };
+            Assert.True(WriteSarifThenReadLevelNode(result) == FailureLevel.Note);
+
+            result = new Result() { Level = FailureLevel.None };
+            Assert.True(WriteSarifThenReadLevelNode(result) == FailureLevel.None);
+
+            result = new Result() { };
+            Assert.True(WriteSarifThenReadLevelNode(result) == null);
+
+            result = new Result() { Level = FailureLevel.Error };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Warning };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration) == FailureLevel.Error);
+
+            result = new Result() { Level = FailureLevel.Error };
+            defaultConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Warning };
+            overrideConfiguration = new ReportingConfiguration() { Enabled = true, Level = FailureLevel.Note };
+            Assert.True(WriteSarifThenReadLevelNode(result, defaultConfiguration, overrideConfiguration) == FailureLevel.Error);
+        }
+
+        [Fact]
+        public void Result_Level_WriteCorrectValue_Warning()
+        {
+            Result result;
+            ReportingConfiguration defaultConfiguration;
+            ReportingConfiguration overrideConfiguration;
+
+            result = new Result() { Level = FailureLevel.Warning };
+            Assert.True(WriteSarifThenReadLevelNode(result) == FailureLevel.Warning);
```

Comment:
>Assert.True(WriteSarifThenReadLevelNode(result) == FailureLevel.Warning); [](http://example.com/codeflow?start=12&length=73)

this test will fail in Main, pass in the level change branch,
because in Main when it is Warning, we do not write the level node in Sarif at all.